File: graphs/c++/adj_list.cpp
#include<bits/stdc++.h>
using namespace std;

void addedge(vector<int> adj_list[], int u, int v){
    adj_list[u].push_back(v);
    adj_list[v].push_back(u);
}

void print_graph(vector<int> adj_list[], int V){
    for(int i=0;i<V;i++){
        cout << i <<'\t';
        for(auto x: adj_list[i]){
            cout << x << '\t';
        }
        printf("\n");
    }
}

int main(){

    int num_vertices = 5;
    vector <int> adj_list[num_vertices];
    addedge(adj_list, 0, 1); 
    addedge(adj_list, 0, 4); 
    addedge(adj_list, 1, 2); 
    addedge(adj_list, 1, 3); 
    addedge(adj_list, 1, 4); 
    addedge(adj_list, 2, 3); 
    addedge(adj_list, 3, 4);

    print_graph(adj_list, num_vertices); 
    return 0;
}

File: graphs/python/adj_list.py

class adj_node: 
    def __init__(self, data):
        self.vertex = data
        self.next = None

class adj_list:
    def __init__(self, num_vertices):
        self.V = num_vertices
        self.list = [None] * self.V
    
    def add_edge(self, src, dest):
        new_node = adj_node(dest)
        new_node.next = self.list[src]
        self.list[src] = new_node

        new_node = adj_node(src)
        new_node.next = self.list[dest]
        self.list[dest] = new_node

    def print_adjlist(self):
        i = 0
        temp = self.list[0]
        while(i<self.V):
            temp = self.list[i]
            if(temp == None):
                break
            print(temp.vertex)
            i+=1

if __name__ == "__main__":
    
    V = 5
    new_graph = adj_list(V)
    new_graph.add_edge(0,1)

    new_graph.print_adjlist()